src/types.rs

```diff
@@ -481,6 +492,398 @@ impl ZipFileData {
+        extra_data_start: Option<u64>,
+        aes_extra_data_start: u64,
+        compression_method: crate::compression::CompressionMethod,
+        aes_mode: Option<(AesMode, AesVendorVersion, CompressionMethod)>,
+        extra_field: Option<Arc<Vec<u8>>>,
+    ) -> Self
+    where
+        S: Into<Box<str>>,
+    {
+        let permissions = options.permissions.unwrap_or(0o100644);
+        let file_name: Box<str> = name.into();
+        let file_name_raw: Box<[u8]> = file_name.bytes().collect();
+        ZipFileData {
+            system: System::Unix,
+            version_made_by: DEFAULT_VERSION,
+            encrypted: options.encrypt_with.is_some(),
+            using_data_descriptor: false,
+            compression_method,
+            compression_level: options.compression_level,
+            last_modified_time: options.last_modified_time,
+            crc32: raw_values.crc32,
+            compressed_size: raw_values.compressed_size,
+            uncompressed_size: raw_values.uncompressed_size,
+            file_name, // Never used for saving, but used as map key in insert_file_data()
+            file_name_raw,
+            extra_field,
+            central_extra_field: options.extended_options.central_extra_data().cloned(),
+            file_comment: String::with_capacity(0).into_boxed_str(),
+            header_start,
+            data_start: OnceLock::new(),
+            central_header_start: 0,
+            external_attributes: permissions << 16,
+            large_file: options.large_file,
+            aes_mode,
+            extra_fields: Vec::new(),
+            extra_data_start,
+            aes_extra_data_start,
+        }
+    }
+
+    pub(crate) fn from_local_block<R: std::io::Read>(
+        block: ZipLocalEntryBlock,
+        reader: &mut R,
+    ) -> ZipResult<Self> {
+        let ZipLocalEntryBlock {
+            // magic,
+            version_made_by,
+            flags,
+            compression_method,
+            last_mod_time,
+            last_mod_date,
+            crc32,
+            compressed_size,
+            uncompressed_size,
+            file_name_length,
+            extra_field_length,
+            ..
+        } = block;
+
+        let encrypted: bool = flags & 1 == 1;
+        let is_utf8: bool = flags & (1 << 1) != 0;
+        let using_data_descriptor: bool = flags & (1 << 3) != 0;
+        #[allow(deprecated)]
+        let compression_method = crate::CompressionMethod::from_u16(compression_method);
+        let file_name_length: usize = file_name_length.into();
+        let extra_field_length: usize = extra_field_length.into();
+
+        if encrypted {
+            return Err(ZipError::UnsupportedArchive(
+                "Encrypted files are not supported",
+            ));
+        }
+        if using_data_descriptor {
+            return Err(ZipError::UnsupportedArchive(
+                "The file length is not available in the local header",
+            ));
+        }
```

Comment:
I have done this, although I would be very surprised if it sped up fuzzing at all since the only instructions in between initializing these variables and checking for validity is a few bit shifts.